guides/specification.md

```diff
@@ -0,0 +1,555 @@
+
+> Execution Scenario:
+> 
+> - 2 targets [`A`, `B`]
+> - selected Checks [`corosync_check`]
+> - some environment (context)
+> 
+> ```yaml
+> expectations:
+>   - name: token_timeout
+>     expect: facts.corosync_token_timeout == values.expected_token_timeout
+> ```
+
+Considering the previous scenario what happens is that:
+- the fact `corosync_token_timeout` is gathered on all targets (`A` and `B` in this case)
+- the expectation expression gets executed against the `corosync_token_timeout` fact gathered on every targets.
+  - `targetA.corosync_token_timeout == values.expected_token_timeout`
+  - `targetB.corosync_token_timeout == values.expected_token_timeout`
+- every evaluation has to be `true`
+
+### Expect Same
+
+This type of expectation is satisfied when, after facts gathering, the expression's return value is the same for all the targets involved in the current execution.
+
+> Execution Scenario:
+> 
+> - 2 targets [`A`, `B`]
+> - selected Checks [`some_check`]
+> - some environment (context)
+> 
+> ```yaml
+> expectations:
+>   - name: awesome_expectation
+>     expect_same: facts.awesome_fact
+> ```
+
+Considering the previous scenario what happens is that:
+- the fact `awesome_fact` is gathered on all targets (`A` and `B` in this case)
+- the expectation expression gets executed for every target involved.
+  - `targetA.facts.awesome_fact`
+  - `targetB.facts.awesome_fact`
+- the expressions results has to be the same for every target
+  - `targetA.facts.awesome_fact == targetB.facts.awesome_fact`
+
+## Expression Language
+
+Different parts of the Check declaration are places where a computation is needed.
+
+> Provide a contextual [Description](#description)/[Remediation](#remediation) text, based on the execution context
+> 
+> interpolation of `{{ <expression> }}`
+```yaml
+description: |
+  Corosync `token` timeout is set to `{{ values.expected_token_timeout }}`
+``` 
+
+> Determine to what a [value](#values) resolves during execution
+> 
+> `when: <expression>` part of a Value's condition
+
+```yaml
+values:
+  - name: expected_token_timeout
+    default: 5000
+    conditions:
+      - value: 30000
+        when: env.provider == "azure" || env.provider == "aws"
+      - value: 20000
+        when: env.provider == "gcp"
+
+```
+
+> Defining the [Expectation](#expectations) of a Check
+>
+> `expect|expect_same: <expression>`
+```yaml
+expectations:
+  - name: token_timeout
+    expect: facts.corosync_token_timeout == values.expected_token_timeout
+```
+
+See [reference for the Expression Language](./expression_language.md).
+
+### Evaluation Scope
+
+Every expression has access to an evaluation scope, allowing to access relevant piece of information to run the expression.
+
+Scopes are namespaced and access to items in the scope is name based.
+
+#### **env**
+
+`env` is a map of information about the context of the running execution, it is set by the system on each execution/check compilation.
+
+Available entries in scope
```

Comment:
I would put this maybe as an example. Wanda doesn't know about which are the coming env values. Actually, it accepts everything coming in the execution request and it depends on the domain logic